src/main/kotlin/com/github/burrunan/s3cache/internal/AwsS3BuildCacheService.kt

```diff
@@ -148,50 +148,72 @@ class AwsS3BuildCacheService internal constructor(
 
     private fun Stopwatch.loadInternal(key: BuildCacheKey, reader: BuildCacheEntryReader): Boolean {
         val bucketPath = key.getBucketPath()
+        logger.info("Loading cache entry '{}' from S3 bucket", bucketPath)
+        val s3Object: ResponseBytes<GetObjectResponse>
         try {
-            logger.info("Loading cache entry '{}' from S3 bucket", bucketPath)
-            s3.getObject(bucketName, bucketPath).use { s3Object ->
-                val contentLength = s3Object.objectMetadata.contentLength
-                if (contentLength > maximumCachedObjectLength) {
+            s3Object = s3.getObject(GetObjectRequest.builder().bucket(bucketName).key(bucketPath).build(), AsyncResponseTransformer.toBytes())
+                    .join()
+        } catch (e : CompletionException) {
+            val cause = e.cause
+            when (cause) {
+                is NoSuchBucketException -> {
+                    logger.info("Bucket '{}' not found", bucketName, cause)
```

Comment:
`NoSuchBucketException` should be rethrown, so Gradle recognizes there's a configuration issue, and it would disable the remote cache.